test/models/pawn_test.rb

```diff
@@ -0,0 +1,121 @@
+    move = pawn.valid_move?([0, 4])
+    assert_not move
+  end
+
+  test "check if move is valid (invalid case, capture vertically)" do
+    game = Game.create(:white_uid => 1, :black_uid => 1, :game_name => "New Game")
+    rook = Rook.create(:pos_x => 0, :pos_y => 2, :active => true, :game_id => game.id)
+    pawn = game.pieces.where(:pos_x => 0).where(:pos_y => 1).first
+    move = pawn.valid_move?([0, 2])
+    assert_not move
+  end
+
+  test "check if move is valid (valid case, starting - capture diagonal - 1 space forward)" do
+    game = Game.create(:white_uid => 1, :black_uid => 1, :game_name => "New Game")
+    rook = Rook.create(:pos_x => 1, :pos_y => 2, :active => true, :game_id => game.id)
+    pawn = game.pieces.where(:pos_x => 0).where(:pos_y => 1).first
+    move = pawn.valid_move?([1, 2])
+    assert move
+  end
+
+  test "check if move is valid (valid case, starting - capture diagonal - 2 spaces forward)" do
+    game = Game.create(:white_uid => 1, :black_uid => 1, :game_name => "New Game")
+    rook = Rook.create(:pos_x => 1, :pos_y => 3, :active => true, :game_id => game.id)
+    pawn = game.pieces.where(:pos_x => 0).where(:pos_y => 1).first
+    move = pawn.valid_move?([1, 3])
+    assert move
+  end
+
+  test "check if move is valid (invalid case, starting - capture diagonal - 3 space forward)" do
+    game = Game.create(:white_uid => 1, :black_uid => 1, :game_name => "New Game")
+    rook = Rook.create(:pos_x => 1, :pos_y => 4, :active => true, :game_id => game.id)
+    pawn = game.pieces.where(:pos_x => 0).where(:pos_y => 1).first
+    move = pawn.valid_move?([1, 4])
+    assert_not move
+  end
+
+  test "check if move is valid (valid case, second move - one space forward)" do
+    game = Game.create(:white_uid => 1, :black_uid => 1, :game_name => "New Game")
+    pawn = game.pieces.where(:pos_x => 0).where(:pos_y => 1).first
+    pawn.update_attributes(:pos_y => 2)
+    move = pawn.valid_move?([0, 3])
+    assert move
+  end
+
+  test "check if move is valid (invalid case, second move - two spaces forward)" do
+    game = Game.create(:white_uid => 1, :black_uid => 1, :game_name => "New Game")
+    pawn = game.pieces.where(:pos_x => 0).where(:pos_y => 1).first
+    pawn.update_attributes(:pos_y => 2)
+    move = pawn.valid_move?([0, 4])
+    assert_not move
+  end
+
+  test "check if move is valid (invalid case, backward move)" do
+    game = Game.create(:white_uid => 1, :black_uid => 1, :game_name => "New Game")
+    pawn = game.pieces.where(:pos_x => 0).where(:pos_y => 1).first
+    pawn.update_attributes(:pos_y => 3)
+    move = pawn.valid_move?([0, 2])
+    assert_not move
+  end
+
+  test "check if move is valid (invalid case, horizontal move)" do
+    game = Game.create(:white_uid => 1, :black_uid => 1, :game_name => "New Game")
+    pawn = game.pieces.where(:pos_x => 0).where(:pos_y => 1).first
+    pawn.update_attributes(:pos_y => 2)
+    move = pawn.valid_move?([1, 2)
```

Comment:
Yeah, that did the trick. Thanks!!

--
Jeff Powers
Global Partners, Inc.
www.globalpartnersinc.com

Connect on Linkedin: http://www.linkedin.com/in/jeffreytpowers/

On Mon, Sep 28, 2015 at 2:27 PM, George Welsted <notifications@github.com>
wrote:

> In test/models/pawn_test.rb
> <https://github.com/Firehose-Team-Awesome/chess-game/pull/26#discussion_r40587253>
> :
>
> > +    assert_not move
> > +  end
> > +
> > +  test "check if move is valid (invalid case, backward move)" do
> > +    game = Game.create(:white_uid => 1, :black_uid => 1, :game_name => "New Game")
> > +    pawn = game.pieces.where(:pos_x => 0).where(:pos_y => 1).first
> > +    pawn.update_attributes(:pos_y => 3)
> > +    move = pawn.valid_move?([0, 2])
> > +    assert_not move
> > +  end
> > +
> > +  test "check if move is valid (invalid case, horizontal move)" do
> > +    game = Game.create(:white_uid => 1, :black_uid => 1, :game_name => "New Game")
> > +    pawn = game.pieces.where(:pos_x => 0).where(:pos_y => 1).first
> > +    pawn.update_attributes(:pos_y => 2)
> > +    move = pawn.valid_move?([1, 2)
>
> looks like you forgot a bracket ]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Firehose-Team-Awesome/chess-game/pull/26/files#r40587253>
> .
>